query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/PinotUtils.java

```diff
@@ -7,4 +7,8 @@ public static String getZkPath(String zkBasePath, String pinotClusterName) {
         ? zkBasePath + pinotClusterName
         : zkBasePath + "/" + pinotClusterName;
   }
+
+  public static String getMetricName(String name) {
+    return name.replaceAll("[^a-zA-Z0-9_]", ".");
```

Comment:
wonder if this should go to the platform-metrics and only allow the alphaNumeric value.

Reply:
Right now, this one is a special use case, other places we are using constant names. yes, but we can consider that. will chat with on this.